### DISCUSSION TITLE
Row selected state resets when data changes.

### DISCUSSION THREAD

**Discussion by avadhutprabhudesai** [TanStack/table#2424] [General]
Hi @tannerlinsley . I have my filtering logic outside of the react-table, I am passing filtered data to the table. However, row select screws up when I perform following steps.

I cannot share code due to confidentiality clauses. Here is a code sandbox link.
https://codesandbox.io/s/spring-cdn-d2yoq
Steps to reproduce:
1. Check checkbox for 2nd row.
2. Enter firstName value of the checked row in the input field above the table.
3. Notice when data is modified, and it becomes the only row in the table, checkbox state screws up.

**Comment by Vasu-p** [comment]
I am having a similar issue where my table's selection is controlled and table selection doesn't properly get set if I try to select a checkbox!
